mtasa-resources/CODING_GUIDELINES.md

```diff
@@ -49,42 +59,92 @@ function exampleFunction(value)
 end
 ```
 
-### Consistent naming conventions
+### Error handling
 
-TODO
+Use the Lua [`error (message [, level])`](https://www.lua.org/manual/5.1/manual.html#pdf-error) function to raise warnings when validating parameters passed to a function, with `level` set to `2` so that the error points the error at the function call location. For example:
 
-### Use of constants
+```lua
+-- This function outputs player's name to the chat box
+function outputPlayerName(player)
+    -- If player is invalid, raise an error
+    if not (isElement(player) and getElementType(player) == "player") then
+        error("Invalid player!", 2)
+    end
+
+    outputChatBox(getPlayerName(player))
+end
 
-TODO
+-- This function is triggered each time a player joins and calls outputPlayerName
+local function playerJoin()
+    outputPlayerName() -- note the missing player argument; the error will point to this line
+end
+addEventHandler("onPlayerJoin", root, playerJoin)
+```
 
-### Indentation and formatting
+### Avoid anonymous functions
 
-Ensure your code editor (e.g. [Visual Studio Code](https://code.visualstudio.com/) 
-applies the rules established by the project's **.editorconfig** file.
+Rather than using anonymous functions (i.e when calling [`addEventHandler`](https://wiki.multitheftauto.com/wiki/AddEventHandler)), use a named local function instead. For example:
 
-# General principles
+```lua
+-- Rather than this:
+addEventHandler("onPlayerConnect", root, function()
+    outputChatBox("Hello!")
+end)
+
+-- Do this:
+local function onConnect()
+    outputChatBox("Hello!")
+end
+addEventHandler("onPlayerConnect", root, onConnect)
+```
 
--   Write clear, readable, and maintainable code.
--   Follow the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
-    (Don't Repeat Yourself) principle.
--   Adhere to the [KISS](https://en.wikipedia.org/wiki/KISS_principle)
-    (Keep It Simple, Stupid) principle.
--   Use meaningful variable and function names that convey their purpose.
--   Comment your code where necessary to explain the logic.
+### Consistent naming conventions
 
-# Script security
+All function names and variables should use the camel case naming convention. Constant variables should use upper snake case (see below).
 
-Follow the [Script security](https://wiki.multitheftauto.com/wiki/Script_security) 
-principles established for MTA:SA scripts to ensure the safety and integrity of your code.
+### Use of constants and avoiding [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming))
+
+Don't use magic numbers in blocks of code - instead, define such values as "constants" at the top of the file using upper snake case.
+
+*Note: the version of Lua used by MTA does not support the `const` keyword added in Lua 5.4. In MTA Lua, the concept of a `const` is just that - a concept.*
```

Comment:
That note is pointless

Reply:
I don't consider it pointless from the perspective of a new contributor who has no knowledge of MTA.

Reply:
To be fair, if someone has no knowledge of MTA then they shouldnt try to contribute until they grasp basics of mta.
Otherwise whats the point?

Reply:
Then surely the more information that is accessible on the subject, the faster they'll be able to "grasp the basics"?

I don't see an issue with this note.

Reply:
Alrighty then